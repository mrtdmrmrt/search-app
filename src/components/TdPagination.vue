<template>
  <div class="td-pagination__wrapper">
    <ul class="td-pagination__list">
      <li class="td-pagination__list--prev" @click="prevPagination">
        Previous
      </li>
      <li
        class="td-pagination__list--item"
        v-for="(page, key) in total"
        :key="key"
        @click="changePagination(page)"
        :class="{ 'active-pagination': page == activePage }"
      >
        {{ page }}
      </li>
      <li class="td-pagination__list--next" @click="nextPagination">Next</li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "TdPagination",
  data() {
    return {
      activePage: 1,
    };
  },
  props: {
    /**
     * data total count
     */
    total: {
      type: Number,
    },
    /**
     * current page
     */
    currentPage: {
      type: Number,
      default: 1,
    },
  },
  watch: {
    currentPage() {
      this.activePage = this.currentPage;
    },
  },
  methods: {
    /**
     * pagination on change
     * @param {number} page
     */
    changePagination(page) {
      this.$emit("on-change-pagination", page);
    },
    prevPagination() {
      if (this.activePage > 1) {
        this.changePagination(--this.activePage);
      }
    },
    nextPagination() {
      if (this.activePage < this.total) {
        this.changePagination(++this.activePage);
      }
    },
  },
  mounted() {
    this.activePage = this.currentPage > this.total ? 1 : this.currentPage;
  },
};
</script>

<style lang="scss">
.td-pagination {
  &__wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &__list {
    list-style: none;
    display: flex;
    font-weight: 900;
    color: #000000;
    font-size: 20px;

    &--item {
      margin: 0 6px;
      cursor: pointer;
      padding: 6px 10px;
      &.active-pagination {
        background: #204080;
        border: 1px solid #ffffff;
        color: #ffffff;
      }
    }

    &--prev {
      cursor: pointer;
      margin-right: 10px;
      padding: 6px 10px;
    }

    &--next {
      cursor: pointer;
      margin-left: 10px;
      padding: 6px 10px;
    }
    & li {
      font-weight: 700;
      font-size: 14px;
      line-height: 16px;

      color: #484848;
      background: #ffffff;
      border: 1px solid #484848;
      border-radius: 4px;
    }
  }
}
</style>
