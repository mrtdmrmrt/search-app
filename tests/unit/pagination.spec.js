import { shallowMount } from "@vue/test-utils";
import TdPagination from "@/components/TdPagination.vue";

describe("TdPagination.vue", () => {
  it("renders props total, currentPage when passed", () => {
    const wrapper = shallowMount(TdPagination, {
      propsData: {
        total: 3,
        currentPage: 1,
      },
    });
    const target = wrapper.find(".active-pagination");

    expect(target.text()).toBe("1");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
