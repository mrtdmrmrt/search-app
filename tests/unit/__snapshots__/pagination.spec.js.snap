// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TdPagination.vue renders props total, currentPage when passed 1`] = `
<div class="td-pagination__wrapper">
  <ul class="td-pagination__list">
    <li class="td-pagination__list--prev">
      Previous
    </li>
    <li class="td-pagination__list--item active-pagination">
      1
    </li>
    <li class="td-pagination__list--item">
      2
    </li>
    <li class="td-pagination__list--item">
      3
    </li>
    <li class="td-pagination__list--next">Next</li>
  </ul>
</div>
`;
